hail/python/hail/vds/variant_dataset.py

```diff
@@ -266,8 +288,10 @@ def error(msg):
         if not isinstance(vd_col_key, hl.tstruct) or len(vd_col_key) != 1 or vd_col_key.types[0] != hl.tstr:
             error(f"expect variant data to have a single col key of type string, found {vd_col_key}")
 
-        if 'END' not in rd.entry or rd.END.dtype != hl.tint32:
-            error("expect field 'END' in entry of reference data with type int32")
+        invalid_end = 'END' not in rd.entry or rd.END.dtype != hl.tint32
+        invalid_len = 'LEN' not in rd.entry or rd.LEN.dtype != hl.tint32
+        if invalid_end and invalid_len:
+            error("expect at least one of 'END' or 'LEN' in entry of reference data with type int32")
```

Comment:
It's probably safer to assert that at least one field exists, and both fields must have type tint32 if they exist. Unless we're careful to fully support a vds with fields e.g. `END: tint32, LEN: tstring`, which I don't think we want to do.

```suggestion
        end_exists = 'END' in rd.entry
        len_exists = 'LEN' in rd.entry
        if not (end_exists or len_exists):
            error("expect at least one of 'END' or 'LEN' in entry of reference data")
        if end_exists and rd.END.dtype != hl.tint32:
            error("'END' field in entry of reference data must have type tint32")
        if len_exists and rd.LEN.dtype != hl.tint32:
            error("'LEN' field in entry of reference data must have type tint32")
```